app/components/DropdownWithModal/index.tsx

```diff
@@ -0,0 +1,111 @@
+import { FC } from "react";
+import Image from "next/image";
+import { Trans } from "@lingui/macro";
+import KeyboardArrowDown from "@mui/icons-material/KeyboardArrowDown";
+
+import { Text } from "@klimadao/lib/components";
+
+import * as styles from "./styles";
+
+interface Item {
+  name: string;
+  icon: StaticImageData;
+  balance?: string;
+  disabled?: boolean;
+}
+
+interface Props {
+  modalTitle: string;
+  label: string;
+  currentItem: Item;
+  items: Item[];
+  isModalOpen: boolean;
+  onItemSelect: (itemName: string) => void;
+  onToggleModal: () => void;
+}
+
+export const DropdownWithModal: FC<Props> = (props) => {
+  const { currentItem } = props;
+  return (
+    <div className={styles.container}>
+      <Text t="caption" color="lightest" className="label">
+        {props.label}
+      </Text>
+      <button onClick={props.onToggleModal}>
+        <div className="start_content">
+          <Image
+            alt={currentItem.name}
+            src={currentItem.icon}
+            width={48}
+            height={48}
+          />
+          <Text t="body2">{currentItem.name}</Text>
+        </div>
+        <div className="end_content">
+          {currentItem.balance && (
+            <Text t="caption" color="lightest">
+              <Trans id="shared.balance">Balance:</Trans>{" "}
+              {Number(props.currentItem.balance).toFixed(5)}{" "}
```

Comment:
I know you haven't got to this part yet but here's how I'd approach it:
1. when /offset view is rendered, and user is connected =>
2.  fetch all the user data on mount. Dispatch a `fetchOffsetData` async thunk action
3. the async thunk action queries the contracts and then dispatches some normal actions to save the balances in `appState.user`
4. access them in this component with `useSelector(selectBalance(props.currentItem))`